### DISCUSSION TITLE
SB Report doesn't show all campaigns

### DISCUSSION THREAD

**Discussion by Awenega** [amzn/ads-advanced-tools-docs#225] [Q&A] [answered]
Hi, I would like to have data on the spending and purchases generated by all my SB campaigns. 
Through V3 reports I can only get the data from SB campaigns with video, but not the data from campaigns with logo + image. 

How can I get the data of all SB campaigns via reports?

**Comment by elizafarley** [comment, marked as answer]
Hi @Awenega, version 3 reports should contain data for all ad types, including product collection, store spotlight, video, and brand video. Any campaigns created after May 2023 (United States) or August 2023 (all other regions) should return data in version 3 reports. The only campaigns **not** returned in version 3 reports are legacy Sponsored Brands campaigns that were built using the now-deprecated version 3 Sponsored Brands APIs. To get data for any legacy campaigns created before the dates above, you will need to continue to use version 2 reporting endpoints.

**Comment by tspicer** [comment]
@elizafarley we have an advertiser that has Sponsored Brand Video campaigns still running since the original creation date in early 2021. However, these campaigns are not showing up in any V2 API Sponsored Brands Reports. They are present in v3 API Sponsored Brand Purchased Product reports, but that report does not have various metrics like impressions. Based on this thread, it is not clear if we should be expecting these to be present and in which API. Is it possible these campaigns are configured in a manner that they may in fact not appear in a report v2?

![Screenshot 2024-03-09 at 11 11 01 AM](https://github.com/amzn/ads-advanced-tools-docs/assets/3750968/49b00b17-5a00-4927-8bd2-94110fdf2abd)

**Comment by elizafarley** [comment]
@tspicer In order to receive video campaign data in version 2 reporting, you need to use the "creativeType": "video" flag in the request. 

For example:

```
curl --request POST 'https://advertising-api.amazon.com/v2/hsa/campaigns/report' \
--header 'Content-Type: application/json' \
--header 'Authorization: Bearer xxxxxx' \
--header 'Amazon-Advertising-API-ClientId: xxxxxx' \
--header 'Amazon-Advertising-API-Scope: xxxxxx' \
--data-raw '{
"reportDate": "20240102",
"metrics": "viewableImpressions,vtr,vctr,clicks,cost,attributedConversions14d,attributedSales14d",
"creativeType": "video"
}'
```

Let me know if that helps.

**Reply by tspicer** [reply]
Hi @elizafarley, thanks for the reply. Yes, we are doing that for v2 campaigns, but these video campaigns for this advertiser are not coming through. I recall there might be an issue related to manual targeting or some setting that may prevent them from coming in. Is this the case? Any other thoughts?

**Reply by elizafarley** [reply]
Can you share an example request body you are using for the version 2 reports call?

**Reply by buraksaler** [reply]
hi @elizafarley. how can I see all the options of creativeType parameter?

**Reply by elizafarley** [reply]
@buraksaler the only valid values for creativeType are `video` and `all`. You can check this in the API spec for [v2/hsa/{recordType}/reports](https://advertising.amazon.com/API/docs/en-us/sponsored-brands/3-0/openapi#tag/Reports/).

**Reply by phase-sys** [reply]
hi @elizafarley, in v3 reporting, would there be a filter for creativeType in the future?

**Reply by adeelsohailahmed** [reply]
Hi @tspicer, did you ever found the solution to your problem? I just encountered the same issue. Video campaigns are not being included in the v2 reports.

**Comment by AmandeepSingh179** [comment]
Hi, @tspicer @adeelsohailahmed  did you ever find a solution to this issue? I’ve just run into the same problem—video campaigns don’t seem to be included in the v2 reports.
